app/controllers/core_induction_programme/modules_controller.rb

```diff
@@ -1,7 +1,35 @@
 # frozen_string_literal: true
 
 class CoreInductionProgramme::ModulesController < ApplicationController
-  def show
-    @course_module = CourseModule.includes(:course_lessons).find(params[:module_id])
+  include Pundit
+  include GovspeakHelper
+
+  after_action :verify_authorized, except: :show
+  before_action :authenticate_user!, except: :show
+  before_action :load_and_authorize_course_module
+
+  def show; end
+
+  def edit
+    @module_preview = params[:module_preview] || @course_module.content
+    @html_content = content_to_html(@module_preview)
+  end
+
+  def update
+    if params[:commit] == "Save changes"
+      @course_module.update!(content: params[:module_preview])
+      redirect_to cip_year_module_url
```

Comment:
So when we redirect we won't even show a message to the user than the update was a success?

Reply:
They will be redirected to the page they just edited, where the success should be easily verifiable.

Are you thinking a flash message? I definitely don't want to add a confirmation page for it.

Reply:
Yes, a flash message exactly. Confirmation pages are best left alone unless the designers demand them.

Reply:
i've done a notice that adds a message on the redirected page.  Are we bothered whether it's an alert or notice? 
<img width="1440" alt="Screenshot 2021-02-03 at 11 43 05" src="https://user-images.githubusercontent.com/69353998/106742680-40a78980-6615-11eb-905b-b009d65fbb23.png">

Reply:
This is a decision for a designer. I expect we don't need this at all, see [this thread](https://ukgovernmentdigital.slack.com/archives/C6DMEH5R6/p1612181773094100?thread_ts=1612181226.088100&cid=C6DMEH5R6)

Reply:
> I kinda think you only need these Success messages if it’s not otherwise obvious from the page. Like, you don’t need one for updating a thing, if the page shows you that updated thing anyway.

Reply:
I've invited Dom to the repo, once she accepts we should be able to tag her on discussions like this. Might be worth doing the same for Mark

Reply:
Thanks for sharing that Ethan. I had a chat with Mark A and he said as long as the message is clear it doesn't matter to much whether it's a notice or alert. I'll include it for now.